### ISSUE TITLE
Language server binary not found: undefined

### ISSUE THREAD

**Issue by aringot** [Exafunction/codeium#36]
Hi, I use VS Code on MacOS and it looks like every time codeium gets an update it downloads the new language server, says it did not find it, and stops working, even though the files are here and everything works after one (sometimes several) restart of VS Code...

Here is an example after restarting VS Code a first time (it was first failing for Codeium 1.6.26, here for 1.6.27):
```
2024-01-29 17:54:18.110 [INFO]: mac-ca initialized
2024-01-29 17:54:19.526 [INFO]: No proxy settings found
2024-01-29 17:54:25.961 [INFO]: Creating manager directory: /var/folders/w2/5hzs1fm560dc7wlhs625z1nm0000gn/T/df017ddf-baba-4eb8-92d0-940f20d49ca4/codeium/manager
2024-01-29 17:54:25.973 [INFO]: 2e835f65b3441bf3f8f401a88a45920578df95d6 not found, downloading language server
2024-01-29 17:54:27.043 [INFO]: No proxy settings found
2024-01-29 17:54:46.229 [ERROR]: Failed to download from https://github.com/Exafunction/codeium/releases/download/language-server-v1.6.27/language_server_macos_x64.gz
2024-01-29 17:54:46.236 [ERROR]: Language server binary not found: undefined
```

When I hover over the (red) Codeium part of the bottom bar it shows an error message saying it could not rename/move the `language_server_macos_x64.download` file to `language_server_macos_x64` but the file is here, in `2e835f65b3441bf3f8f401a88a45920578df95d6/language_server_macos_x64`.

Again, it finally works if I restart again... But that is obviously a bug.

**Comment by vreoo** [comment]
I have Disabled/Enabled the Extension multiple times and Uninstalled/Installed the extension various times too, It downloads to a certain point then it stops and throws the same error in the Logs output:
```
2024-05-26 10:16:30.857 [INFO]: mac-ca initialized
2024-05-26 10:16:31.290 [INFO]: No proxy settings found
2024-05-26 10:16:35.142 [INFO]: Creating manager directory: /var/folders/h3/k4byl0m94lg2lf6qm2smbxfh0000gp/T/67723bc7-6ebe-485a-8888-782bafba78c7/codeium/manager
2024-05-26 10:16:35.142 [INFO]: 83dadda2bd1a9946542096cce0cfd2bb4d2a9074 not found, downloading language server
2024-05-26 10:16:35.157 [INFO]: No proxy settings found
2024-05-26 10:17:04.013 [ERROR]: Failed to download from https://github.com/Exafunction/codeium/releases/download/language-server-v1.8.45/language_server_macos_arm.gz
2024-05-26 10:17:04.024 [ERROR]: Language server binary not found: undefined
```

I am not able to use Codium for now because of this issue. Is there any way I can download the language server manually at least?

<be />

**UPDATE:** Finally it downloaded the language server after I disabled autocomplete, then disabled the extension and re-enabled it again so it worked

I hope this helps

**Comment by concypt** [comment]
@vreoo not working for me this is what in my output 

`(VSCode) 2024-08-14 18:21:13.127 [INFO]: mac-ca initialized
(VSCode) 2024-08-14 18:21:13.137 [INFO]: No proxy settings found
(VSCode) 2024-08-14 18:21:14.006 [INFO]: Creating manager directory: /var/folders/v7/qq603tc976gg9njvs4zfj5780000gn/T/ce0a8d00-9c1c-4e42-b756-9cf9c063eaf2/codeium/manager
(VSCode) 2024-08-14 18:21:14.008 [INFO]: b7cd5e218f7791eb6d2930237ba6b7651a72b479 not found, downloading language server
(VSCode) 2024-08-14 18:21:14.027 [INFO]: No proxy settings found
(VSCode) 2024-08-14 18:21:17.962 [ERROR]: Failed to download from https://github.com/Exafunction/codeium/releases/download/language-server-v1.12.1/language_server_macos_arm.gz
(VSCode) 2024-08-14 18:21:17.965 [ERROR]: Language server binary not found: undefined`

**Comment by vreoo** [comment]
@concypt Did you do the steps I mentioned previously?

> UPDATE: Finally it downloaded the language server after I disabled autocomplete, then disabled the extension and re-enabled it again so it worked

**Comment by concypt** [comment]
@vreoo Yes, but it does not make the situation better. I have done it multiple times but not worked. Finally, I uninstalled the extension and reinstalled it a few times and it worked. I still do not know the reason because I do not have any firewall or found anything that could be the issue.
